modules/renter/downloadchunk.go

```diff
@@ -0,0 +1,149 @@
+package renter
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+
+	"github.com/NebulousLabs/errors"
+)
+
+// downloadPieceInfo contains all the information required to download and
+// recover a piece of a chunk from a host. It is a value in a map where the key
+// is the file contract id.
+type downloadPieceInfo struct {
+	index uint64
+	root  crypto.Hash
+}
+
+// unfinishedDownloadChunk contains a chunk for a download that is in progress.
+//
+// TODO: Explore making workersStandby a heap sorted by latency or whatever
```

Comment:
I'd be happy to tear out the `TODO` statements and turn them into GH Issues.  If these get merged in I can start to create issues from them.

Reply:
I definitely use the TODO statements when I return to the code months later. I want to make sure that every idea that I've had which is potentially worthwhile is documented somewhere, and the TODOs I feel are a pretty good way to do that.

I'm also in full support of taking some of the TODOs and turning them into GH issues for people to tackle. I wouldn't want to overwhelm people, so maybe only 2-5 per package at a time, but as people tackle the ones we've turned into issues, we can add the remaining ones as issues as well.

Reply:
That would be great. It feels like there are dozens of TODOs and some of them are low-hanging fruits that could easily be tackled by the community. It's way too easy to lose track of them if there is no central organized list of TODOs because they are scattered over hundreds of files.
I wouldn't mind having the TODOs in the code as long as we also have an issue linking to that todo.